src/ts/client/grapesjs/keymaps.ts

```diff
@@ -0,0 +1,111 @@
+  const panels: string[] = [
+    cmdToggleBlocks,
+    cmdToggleLayers,
+    cmdToggleNotifications,
+    cmdToggleSymbols,
+    cmdTogglePages,
+    cmdOpenSettings,
+    cmdOpenFonts
+  ]
+  panels.forEach(p => editor.Commands.stop(p))
+}
+
+/**
+ * Escapes the current context in this order : modal, Publish dialog, left panel.
+ * @param editor The editor.
+ */
+const escapeContext = (editor: Editor): void => {
+  const publishDialog = (editor as PublishableEditor).PublicationManager.dialog
+
+  if (editor.Modal.isOpen()) {
+    console.log('closing modal')
+    editor.Modal.close()
+  } else if (publishDialog && publishDialog.isOpen) {
+    publishDialog.closeDialog()
+  } else {
+    resetPanel(editor)
+  }
+}
+
+// Main part
+
+export const keymapsPlugin = (editor: Editor, opts: PluginOptions): void => {
+  // Commands
+  editor.Commands.add(cmdSelectBody, selectBody)
+
+  if (opts.disableKeymaps) return
+  if (opts.prefixKey) prefixKey = opts.prefixKey
+
+  const km = editor.Keymaps
+
+  // Panels
+  km.add('general:open-settings', defaultKms.kmOpenSettings.toLowerCase(), editor => toggleCommand(editor, cmdOpenSettings))
+  km.add('general:open-publish', defaultKms.kmOpenPublish.toLowerCase(), editor => toggleCommand(editor, cmdPublish))
+  km.add('general:open-fonts', defaultKms.kmOpenFonts.toLowerCase(), editor => toggleCommand(editor, cmdOpenFonts))
+  km.add('general:preview-mode', defaultKms.kmPreviewMode.toLowerCase(), editor => toggleCommand(editor, 'preview'), {prevent: true})
+  km.add('panels:layers', defaultKms.kmLayers.toLowerCase(), editor => toggleCommand(editor, cmdToggleLayers))
+  km.add('panels:blocks', defaultKms.kmBlocks.toLowerCase(), editor => toggleCommand(editor, cmdToggleBlocks))
+  km.add('panels:notifications', defaultKms.kmNotifications.toLowerCase(), editor => toggleCommand(editor, cmdToggleNotifications))
+  km.add('panels:pages', defaultKms.kmPages.toLowerCase(), editor => toggleCommand(editor, cmdTogglePages))
+  km.add('panels:symbols', defaultKms.kmSymbols.toLowerCase(), editor => toggleCommand(editor, cmdToggleSymbols))
+  km.add('panels:close-panel', defaultKms.kmClosePanel.toLowerCase(), escapeContext)
+
+  // Handling the Escape keymap during text edition
+  document.addEventListener('keydown', event => {
+    if (event.key === defaultKms.kmClosePanel) {
+      const target = event.target as HTMLElement | null
+      const rte = editor.getEditing()
+
+      if (rte) { // If in Rich Text edition...
+        // TODO: Close the Rich Text edition and un-focus the text field
+      } else if (target) { // If target exists...
+        if (target.tagName === 'INPUT' && target.getAttribute('type') === 'submit') { // If it's a submit button...
+          escapeContext(editor)
+        } else if (isTextOrInputField(target)) { // If it's a text field...
+          target.blur()
+        }
+      }
+    }
+  })
+
+  // Workflow-specific keymaps
+  km.add('workflow:select-body', prefixKey + '+b', cmdSelectBody)
```

Comment:
why not make it `esc` to select the body? maybe only when the dialog stack is empty

Reply:
I think Jonas wanted a shortcut for this when he's editing HTML attributes, he's doing a back-and-forth. I can add this, but it seems a bit unintuitive (in my opinion) and not consistent, since it won't work when we have a left panel (which is often the case, I imagine?), so it will add friction. Tell me what's your verdict!

Reply:
The back and forth is a workaround because there is a bug in some expression inputs, so don't take it into account, I will fix the bug instead

I think esc for selecting the body is good but only when we are not in a rich text 

Maybe we can try and test it?

Reply:
I can try, sure. I'm considering we that we also keep ``Shift+B`` (?)